cloud/deploy/deploy.go

```diff
@@ -117,6 +125,7 @@ func deployDags(path, runtimeID string, client astro.Client) error {
 	// Delete the tar file
 	defer func() {
 		dagFile.Close()
+		os.Remove(monitoringDagPath)
```

Comment:
nitpick: not directly related to this, but shouldn't the defer call be defined right after the operation we want to revert at the end of function execution, else in case the function fails because of an error in `ReportDagDeploymentStatus` or `azureUploader` the tar file as well as the monitoring dag both won't be deleted i.e.

```Golang
        dagDeployment, err := deployment.Initiate(runtimeID, client)
	if err != nil {
		return err
	}

	// Check the dags directory
	dagsPath := path + "/dags"
	monitoringDagPath := filepath.Join(dagsPath, "astronomer_monitoring_dag.py")

	// Create monitoring dag file
	err = fileutil.WriteStringToFile(monitoringDagPath, include.MonitoringDag)
	if err != nil {
		return err
	}

	// Generate the dags tar
	err = fileutil.Tar(dagsPath, path)
	if err != nil {
		return err
	}

         // Delete the tar file
	defer func() {
		os.Remove(monitoringDagPath)
		err = os.Remove(dagFile.Name())
		if err != nil {
			fmt.Println("\nFailed to delete dags tar file: ", err.Error())
			fmt.Println("\nPlease delete the dags tar file manually from path: " + dagFile.Name())
		}
	}()
			
	dagsFilePath := path + "/dags.tar"
	dagFile, err := os.Open(dagsFilePath)
	if err != nil {
		return err
	}
	defer dagFile.Close()
	versionID, err := azureUploader(dagDeployment.DagURL, dagFile)
	if err != nil {
		return err
	}
	var status string
	if versionID != "" {
		status = "SUCCEEDED"
	} else {
		status = "FAILED"
	}
	_, err = deployment.ReportDagDeploymentStatus(dagDeployment.ID, runtimeID, action, versionID, status, message, client)
	if err != nil {
		return err
	}
```

Reply:
Will need to do it post uploading the file.